File: src/test/java/com/demonwav/mcdev/MinecraftCodeInsightFixtureTestCase.java
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2016 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev;

import com.demonwav.mcdev.platform.MinecraftModuleType;
import com.demonwav.mcdev.platform.bukkit.PaperModuleType;
import com.demonwav.mcdev.platform.bungeecord.BungeeCordModuleType;
import com.demonwav.mcdev.platform.canary.NeptuneModuleType;
import com.demonwav.mcdev.platform.forge.ForgeModuleType;
import com.demonwav.mcdev.platform.liteloader.LiteLoaderModuleType;
import com.demonwav.mcdev.platform.mcp.McpModuleType;
import com.demonwav.mcdev.platform.mixin.MixinModuleType;
import com.demonwav.mcdev.platform.sponge.SpongeModuleType;

import com.intellij.openapi.module.Module;
import com.intellij.openapi.roots.ContentEntry;
import com.intellij.openapi.roots.ModifiableRootModel;
import com.intellij.testFramework.LightProjectDescriptor;
import com.intellij.testFramework.fixtures.LightCodeInsightFixtureTestCase;
import org.jetbrains.annotations.NotNull;

public class MinecraftCodeInsightFixtureTestCase extends LightCodeInsightFixtureTestCase {
    @NotNull
    @Override
    protected LightProjectDescriptor getProjectDescriptor() {
        return new LightProjectDescriptor() {
            @Override
            protected void configureModule(@NotNull Module module, @NotNull ModifiableRootModel model, @NotNull ContentEntry contentEntry) {
                // enable everything
                MinecraftModuleType.addOption(module, PaperModuleType.getInstance().getId());
                MinecraftModuleType.addOption(module, SpongeModuleType.getInstance().getId());
                MinecraftModuleType.addOption(module, ForgeModuleType.getInstance().getId());
                MinecraftModuleType.addOption(module, LiteLoaderModuleType.getInstance().getId());
                MinecraftModuleType.addOption(module, BungeeCordModuleType.getInstance().getId());
                MinecraftModuleType.addOption(module, MixinModuleType.getInstance().getId());
                MinecraftModuleType.addOption(module, McpModuleType.getInstance().getId());
                MinecraftModuleType.addOption(module, NeptuneModuleType.getInstance().getId());
            }
        };
    }
}

File: src/test/kotlin/com/demonwav/mcdev/MinecraftCodeInsightFixtureTestCase.kt
package com.demonwav.mcdev

import com.demonwav.mcdev.platform.MinecraftModuleType
import com.demonwav.mcdev.platform.bukkit.PaperModuleType
import com.demonwav.mcdev.platform.bungeecord.BungeeCordModuleType
import com.demonwav.mcdev.platform.canary.NeptuneModuleType
import com.demonwav.mcdev.platform.forge.ForgeModuleType
import com.demonwav.mcdev.platform.liteloader.LiteLoaderModuleType
import com.demonwav.mcdev.platform.mcp.McpModuleType
import com.demonwav.mcdev.platform.mixin.MixinModuleType
import com.demonwav.mcdev.platform.sponge.SpongeModuleType
import com.intellij.openapi.module.Module
import com.intellij.openapi.roots.ContentEntry
import com.intellij.openapi.roots.ModifiableRootModel
import com.intellij.testFramework.LightProjectDescriptor
import com.intellij.testFramework.fixtures.LightCodeInsightFixtureTestCase

abstract class MinecraftCodeInsightFixtureTestCase : LightCodeInsightFixtureTestCase() {
    override fun getProjectDescriptor(): LightProjectDescriptor {
        return object : LightProjectDescriptor() {
            override fun configureModule(module: Module, model: ModifiableRootModel, contentEntry: ContentEntry) {
                super.configureModule(module, model, contentEntry)
                // enable everything
                MinecraftModuleType.addOption(module, PaperModuleType.getInstance().id)
                MinecraftModuleType.addOption(module, SpongeModuleType.getInstance().id)
                MinecraftModuleType.addOption(module, ForgeModuleType.getInstance().id)
                MinecraftModuleType.addOption(module, LiteLoaderModuleType.getInstance().id)
                MinecraftModuleType.addOption(module, BungeeCordModuleType.getInstance().id)
                MinecraftModuleType.addOption(module, MixinModuleType.getInstance().id)
                MinecraftModuleType.addOption(module, McpModuleType.getInstance().id)
                MinecraftModuleType.addOption(module, NeptuneModuleType.getInstance().id)
            }
        }
    }
}